flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -806,6 +810,19 @@ public void createTable(
                                     ignoreIfExists);
 
                     catalog.createTable(path, resolvedListenedTable, ignoreIfExists);
+                    if (table instanceof CatalogTable) {
```

Comment:
Is there any specific reason to check the type of `table` here? The logic below directly uses `resolvedTable` to create a `CreateTableEvent`, so I'm wondering if it's necessary to have the check here.

Reply:
Thanks @PatrickRen , you are right, it needs to maintain the same object between `instanceof` and table event. I have updated all `createTable`/`alterTable`/`dropTable` methods for table and temporal table. Also I found there is an issue for temporal table and view, I have fixed it in the new commit, thanks